cfgov/v1/jinja2tags/__init__.py

```diff
@@ -98,6 +98,34 @@ def render_stream_child(context, stream_child):
     return Markup(unescaped)
 
 
+def unique_id_in_context(context):
+    """For a given request, return a unique ID each time this method is
+    called. The goal is to generate IDs to uniquely identify input fields
+    in a template that are consistent between page loads.
+
+    If the context has a request object, the generated id will increment:
+
+    >>> context = {'request': request}
+    >>> unique_id_in_context(context)  # returns 0
+    >>> unique_id_in_context(context)  # returns 1
+    >>> unique_id_in_context(context)  # returns 2
+
+    If the context lacks a request, this function will return a 14-character
+    unique alphanumeric string.
+    """
+    request = context.get('request')
+    if request:
+        attribute_name = '__unique_input_ids'
+        if not hasattr(request, attribute_name):
+            setattr(request, attribute_name, 0)
+        id_count = getattr(request, attribute_name)
+        setattr(request, attribute_name, id_count + 1)
```

Comment:
I would expect that the request would contain the last number used (for example, 0 if one call, 1 if two calls, etc). This might be helpful to debug any issues because at any moment the request object would represent the last ID used, although in that case I might expect the attribute name to be something like `__last_unique_id`.

If you'd prefer to leave the logic as is, could we change the attribute name to something like `__next_unique_id`?

It seems like we don't necessarily need to include the word `input` in the attribute name (or mention "input fields" in the docstring above) since in theory one might want to use this template tag on some other kind of element needing a unique ID (like a `<div>`).